sdk/identity/azure-identity/src/managed_identity_credential.cpp

```diff
@@ -20,16 +20,20 @@ std::unique_ptr<_detail::ManagedIdentitySource> CreateManagedIdentitySource(
          AzureArcManagedIdentitySource::Create,
          ImdsManagedIdentitySource::Create};
 
-  for (auto create : managedIdentitySourceCreate)
+  // IMDS ManagedIdentity, which comes last in the list, will never return nullptr from Create().
+  // For that reason, it is not possible to cover that execution branch in tests.
+  for (auto create : managedIdentitySourceCreate) // LCOV_EXCL_LINE
   {
     if (auto source = create(clientId, options))
     {
       return source;
     }
   }
 
+  // LCOV_EXCL_START
   throw AuthenticationException(
       "ManagedIdentityCredential authentication unavailable. No Managed Identity endpoint found.");
```

Comment:
If this will never happen, should we throw here or terminate with `AzureNoReturnPath`?

Reply:
Maybe, I tend to like exceptions more, but maybe we should think about using the `_azure_UNREACHABLE_CODE()`. But I'd do it in a separate PR.

Reply:
#3185